exercises/metric_conversion.livemd

```diff
@@ -96,9 +96,26 @@ defmodule Metric do
     iex> Metric.convert({:kilometer, 10}, :centimeter)
     1000000
   """
-  def convert(from, to) do
+  def convert({unit, amount}, to) do
+    case {unit, amount} do
+      {:kilometer, _} -> convert_to(amount * 1000, to)
```

Comment:
You could simplify this case statement, since you're ignoring the second element in the tuple.

```elixir
case unit do
  :kilometer -> convert_to(amount * 1000, to)
end
```

You could also bind the amount in the case statement to avoid calling `convert_to/2` repetitively.

```elixir
meters = case unit do
   :kilometer -> amount * 1000
end

convert_to(meters, to)
```

You might consider abstracting the functions to convert to and from meters.

```elixir
def to_meters(unit, amount) do
end

def from_meters(unit, amount) do
end
```

Perhaps the magic variables could also be named module attributes?
And/or You could use Icia's magical map to make a map with the conversion values.

Reply:
Ok @BrooklinJazz , this is epic.

Asynchronous support works :)

What is a magic variable?

Thanks for the suggestions. I have pushed another solution...

Reply:
Very happy you're enjoying the review process! I've been inspired to include this for future students.

A magic variable is a value in the codebase that has no context and is not bound to a named variable or module attribute. Sometimes this is called "magic numbers" or "magic strings"

For example:

```elixir
100 * 1.14
```

What is `100` and `1.14` in this code? there's no context. But the following makes more sense:

```elixir
order_total = 100
tax = 1.14

order_total * tax
```